src/vulkan/vk-acceleration-structure.cpp

```diff
@@ -117,10 +122,20 @@ Result AccelerationStructureBuildDescConverter::convert(
     }
     case AccelerationStructureBuildInputType::Triangles:
     {
+        bool useMotion = is_set(buildDesc.flags, AccelerationStructureBuildFlags::CreateMotion);
+        if (useMotion)
+        {
+            motionTrianglesDatas.resize(buildDesc.inputCount);
+        }
+
         for (uint32_t i = 0; i < buildDesc.inputCount; ++i)
         {
             const AccelerationStructureBuildInputTriangles& triangles = buildDesc.inputs[i].triangles;
-            if (triangles.vertexBufferCount != 1)
+            if (triangles.vertexBufferCount > kMaxAccelerationStructureMotionKeyCount)
```

Comment:
For Vulkan I think the only valid case when `useMotion == true` is `vertexBufferCount == 2`. This sort of happens now because `kMaxAccelerationStructureMotionKeyCount == 2` but that could be lifted for OptiX if we really wanted to.